### PR TITLE
feat(sto): improve offering details

### PR DESCRIPTION
**Pull request by monitz87**
Added total offering amount, total remaining tokens and refined status

BREAKING CHANGE: Sto status is now divided into `timing`, `sale` and `balance`

### CODE REVIEW

File: src/api/entities/SecurityToken/Offerings.ts [hunk 1]
```diff
@@ -46,9 +46,9 @@ export class Offerings extends Namespace<SecurityToken> {
   /**
    * Retrieve all of the Token's Offerings. Can be filtered using parameters
    *
-   * @param opts.status - status of the offerings to fetch
+   * @param opts.status - status of the offerings to fetch. As long as the STO has one of the passed statuses, it will be returned
```
Comment (VictorVicente) [inline src/api/entities/SecurityToken/Offerings.ts:49]: This comment is not accurate. STO must have all the passed statuses to be returned.

File: src/api/procedures/investInSto.ts [hunk 2]
```diff
@@ -110,25 +118,21 @@ export async function prepareInvestInSto(
...
-    throw new PolymeshError({
-      code: ErrorCode.ValidationError,
-      message: 'The STO is not live',
-    });
-  }
-
-  const now = new Date();
-
-  if (end && now > end) {
+  if (sale !== StoSaleStatus.Live || timing !== StoTimingStatus.Started) {
```
Comment (VictorVicente) [inline src/api/procedures/investInSto.ts:132]: Should `balanceStatus` be `Available` too?

File: src/api/entities/SecurityToken/Offerings.ts (hunk 1 above)
Comment (monitz87) [reply src/api/entities/SecurityToken/Offerings.ts:49]: I had this in place before we spoke, I forgot to change it 😅

File: src/api/procedures/investInSto.ts (hunk 2 above)
Comment (monitz87) [reply src/api/procedures/investInSto.ts:132]: Agreed

Comment (monitz87) [reply src/api/procedures/investInSto.ts:132]: Wait, that is already considered further down. The reason it's calculated apart and not derived from the status is the `maxPrice` parameter

**Review by VictorVicente — APPROVED** [review]

**Pull request merged**